packages/lodestar/test/unit/chain/attestation.test.ts

```diff
@@ -89,28 +94,32 @@ describe("AttestationProcessor", function() {
       const state = generateState();
       dbStub.state.get.resolves(state);
       forkChoiceStub.getJustified.returns({})
-  
+
       await attestationProcessor.processAttestation(attestation, attestationHash);
       fail("expect an AssertionError");
-    } catch(err) {
+    } catch (err) {
       expect(getAttestingIndicesStub.called).to.be.false;
       expect(forkChoiceStub.addAttestation.called).to.be.false;
     }
   });
 
-  it("processAttestation - should call forkChoice", async () => {
-      const attestation = generateEmptyAttestation();
-      const attestationHash = hashTreeRoot(config.types.Attestation, attestation);
-      const block = generateEmptySignedBlock();
-      dbStub.block.get.resolves(block);
-      const state = generateState();
-      dbStub.state.get.resolves(state);
-      forkChoiceStub.getJustified.returns({});
-      getAttestingIndicesStub.returns([0]);
-      state.balances = [1n, 2n, 3n];
-  
+  it.only("processAttestation - should call forkChoice", async () => {
+    const attestation = generateEmptyAttestation();
+    const attestationHash = hashTreeRoot(config.types.Attestation, attestation);
+    const block = generateEmptySignedBlock();
+    dbStub.block.get.resolves(block);
+    const state = generateState();
+    dbStub.state.get.resolves(state);
+    forkChoiceStub.getJustified.returns({});
+    getAttestingIndicesStub.returns([0]);
+    state.balances = [];
+    state.validators = generateValidators(3, {})
+
+    await timeout(config.params.SECONDS_PER_SLOT)
```

Comment:
you already have method called sleep. That sait, why are we waiting here (we ain't suppose to do that in unit tests)?

you can mock time with sinon (checkout tests in attestation collection PR)

Reply:
I originally waited here because [this assertion](would fail) since the test's generated state's `genesisTime` wouldn't be `SECONDS_PER_SLOT` seconds earlier than the attestation's slot. To work around that, I just let that amount of time pass to satisfy the assertion.

Reply:
Now that you mention it, though, I can just add `SECONDS_PER_SLOT` seconds to the generated state's `genesisTime` and it should work.